src/Chocolatey.PowerShell/Commands/InstallChocolateyPathCommand.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright © 2017 - 2024 Chocolatey Software, Inc
+// Copyright © 2011 - 2017 RealDimensions Software, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Chocolatey.PowerShell.Helpers;
+using Chocolatey.PowerShell.Shared;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Chocolatey.PowerShell.Commands
+{
+    [Cmdlet(VerbsLifecycle.Install, "ChocolateyPath")]
+    [OutputType(typeof(void))]
+    public class InstallChocolateyPathCommand : ChocolateyCmdlet
+    {
+        [Parameter(Mandatory = true, Position = 0)]
+        [Alias("PathToInstall")]
+        public string Path { get; set; }
+
+        [Parameter(Position = 1)]
+        [Alias("Scope", "Type")]
+        public EnvironmentVariableTarget PathType { get; set; } = EnvironmentVariableTarget.User;
+
+        protected override void EndProcessing()
+        {
+            Paths.InstallPathEntry(this, Path, PathType);
+            base.EndProcessing();

```

Comment:
We discussed this earlier today. Has to do with how the logging was or wasn't applied on the original PowerShell commands these were rewritten from. The _usual_ rule of thumb seems to be that public API commands do the logging stuff I've put into the base class's Begin/EndProcessing methods, and non-public API commands don't.

Possibly it might make sense to have a secondary base class that _just_ provides the logging bits for everything else to use? 🤔 Didn't seem worth it when I was initially writing it, but. Maybe.

Reply:
It does communicate it's more intentional, but I'm not sure it actually communicates _why_ very well either IMO. :/

Reply:
I think I have a bit of a nicer way to do this, I'll push changes for this shortly.